pkg/pgmodel/sql_ingest.go

```diff
@@ -522,34 +576,48 @@ func tryRecovery(conn pgxConn, req copyRequest, err error) error {
 	return err
 }
 
-func doInsert(conn pgxConn, req copyRequest) (err error) {
-	numRows := 0
-	for i := range req.data.batch.sampleInfos {
-		numRows += len(req.data.batch.sampleInfos[i].samples)
-	}
-	times := make([]time.Time, 0, numRows)
-	vals := make([]float64, 0, numRows)
-	series := make([]int64, 0, numRows)
-	for req.data.batch.Next() {
-		time, val, serie := req.data.batch.Values()
-		times = append(times, time)
-		vals = append(vals, val)
-		series = append(series, int64(serie))
-	}
-	if len(times) != numRows {
-		panic("invalid insert request")
+func doInsert(conn pgxConn, reqs ...copyRequest) (err error) {
+	batch := conn.NewBatch()
+	numRowsPerInsert := make([]int, 0, len(reqs))
+	for r := range reqs {
+		req := &reqs[r]
+		numRows := 0
+		for i := range req.data.batch.sampleInfos {
+			numRows += len(req.data.batch.sampleInfos[i].samples)
+		}
+		times := make([]time.Time, 0, numRows)
+		vals := make([]float64, 0, numRows)
+		series := make([]int64, 0, numRows)
+		for req.data.batch.Next() {
+			time, val, serie := req.data.batch.Values()
+			times = append(times, time)
+			vals = append(vals, val)
+			series = append(series, int64(serie))
+		}
+		if len(times) != numRows {
+			panic("invalid insert request")
```

Comment:
Should we return error than panicking? The user might not be expecting a panic here.

Reply:
if this occurs it means the number of timestamps we're trying to insert do not equal the number of datapoints we're trying to insert. The only way that can occur is if there's an internal error in the connector, in which case shutting it down before it can cause any more damage is the best thing to do.

Reply:
Agreed. Just meant for a smoother error handling.